crypto/mod.ts

```diff
@@ -0,0 +1,247 @@
+    ): Promise<ArrayBuffer>;
+
+    generateKey?(
+      algorithm: WebCryptoAlgorithmIdentifier,
+      extractable: boolean,
+      keyUsages: string[],
+    ): Promise<unknown>;
+    deriveKey?(
+      algorithm: WebCryptoAlgorithmIdentifier,
+      baseKey: unknown,
+      derivedKeyType: string,
+      extractable: boolean,
+      keyUsages: string[],
+    ): Promise<unknown>;
+    deriveBits?(
+      algorithm: WebCryptoAlgorithmIdentifier,
+      baseKey: unknown,
+      length: number,
+    ): Promise<unknown>;
+
+    importKey?(
+      format: string,
+      keyData: BufferSource | unknown,
+      algorithm: WebCryptoAlgorithmIdentifier,
+      extractable: boolean,
+      keyUsages: string[],
+    ): Promise<unknown>;
+    exportKey?(format: string, key: unknown): Promise<unknown>;
+
+    wrapKey?(
+      format: string,
+      key: unknown,
+      wrappingKey: unknown,
+      wrappingAlgorithm: WebCryptoAlgorithmIdentifier,
+    ): Promise<unknown>;
+    unwrapKey?(
+      format: string,
+      wrappedKey: BufferSource,
+      unwrappingKey: unknown,
+      unwrapAlgorithm: WebCryptoAlgorithmIdentifier,
+      unwrappedKeyAlgorithm: WebCryptoAlgorithmIdentifier,
+      extractable: boolean,
+      keyUsages: string[],
+    ): Promise<unknown>;
+  };
+}
+
+/**
+ * A copy of the global WebCrypto interface, with methods bound so they're
+ * safe to re-export.
+ */
+const webCrypto: WebCrypto = ((crypto) => ({
+  getRandomValues: crypto.getRandomValues?.bind(crypto),
+  randomUUID: crypto.randomUUID?.bind(crypto),
+  subtle: {
+    decrypt: crypto.subtle?.decrypt?.bind(crypto.subtle),
+    deriveBits: crypto.subtle?.deriveBits?.bind(crypto.subtle),
+    deriveKey: crypto.subtle?.deriveKey?.bind(crypto.subtle),
+    digest: crypto.subtle?.digest?.bind(crypto.subtle),
+    encrypt: crypto.subtle?.encrypt?.bind(crypto.subtle),
+    exportKey: crypto.subtle?.exportKey?.bind(crypto.subtle),
+    generateKey: crypto.subtle?.generateKey?.bind(crypto.subtle),
+    importKey: crypto.subtle?.importKey?.bind(crypto.subtle),
+    sign: crypto.subtle?.sign?.bind(crypto.subtle),
+    unwrapKey: crypto.subtle?.unwrapKey?.bind(crypto.subtle),
+    verify: crypto.subtle?.verify?.bind(crypto.subtle),
+    wrapKey: crypto.subtle?.wrapKey?.bind(crypto.subtle),
+  },
+}))(globalThis.crypto as WebCrypto);
+
+const bufferSourceBytes = (data: BufferSource | unknown) => {
+  let bytes: Uint8Array | undefined;
+  if (data instanceof Uint8Array) {
+    bytes = data;
+  } else if (ArrayBuffer.isView(data)) {
+    bytes = new Uint8Array(
+      data.buffer,
+      data.byteOffset,
+      data.byteLength,
+    );
+  } else if (data instanceof ArrayBuffer) {
+    bytes = new Uint8Array(data);
+  }
+  return bytes;
+};
+
+/**
+ * An wrapper for WebCrypto adding support for additional non-standard
+ * algorithms, but delegating to the runtime WebCrypto implementation whenever
+ * possible.
+ */
+const stdCrypto = (<T extends WebCrypto>(x: T) => x)({
```

Comment:
My original objection to dropping the JavaScript/TypeScript implementations in lieu of the existing web assembly ones was that the existing web assembly ones didn't support signing specifically with HMAC, while the JavaScript implementations did, and I have a project (oak) that uses signing.  Of course now that we have WebCrypto, for the 4 web crypto algorithms, we would still have signing with HMAC (and the other types supported by web crypto) but it seems like we should try to support at least signing as part of this.  It doesn't have to be "now" but I think it should be before we drop any of the JavaScript/TypeScript implementations that are not covered by Web Crypto.

Reply:
That sounds like a good plan. I don't want to try to take on the signing as part of this PR, but it would be nice to get it done soon.